File: ex4/main2.rb
# frozen_string_literal: true

def partial_overlap first, second
  return false if (first[1] < second[0] && first[0] < second[0]) || (first[0] > second[1] && first[1] > second[1])

  true
end

file_path = 'input.txt'
file = File.open(file_path)
data = file.read.split("\n")
ans = 0
data.each do |line|
  line = line.split(',')
  first = line[0].split('-').map(&:to_i)
  second = line[1].split('-').map(&:to_i)
  ans += 1 if partial_overlap first, second
end
puts ans

File: Other/ex13/main2.py
def compare(list1, list2):
    if type(list1) is int and type(list2) is int:
        if list1 < list2: return True
        elif list1 > list2: return False
        else : return None
    else:
        if type(list1) is int:
            list1 = [list1]
        if type(list2) is int:
            list2 = [list2]
        for i in range(len(list1)):
            if i >= len(list2):
                return False
            if compare(list1[i], list2[i]) is not None:
                return compare(list1[i], list2[i])
        if len(list1) < len(list2):
            return True
        else:
            return None

def sort(lists):
    for i in range(len(lists)):
        for j in range(i + 1, len(lists)):
            if not compare(lists[i], lists[j]):
                lists[i], lists[j] = lists[j], lists[i]

def main():
    with open('input.txt', 'r') as f:
        lines = f.read().split("\n")

    lists = [[[2]],[[6]]]

    for line in lines:
        if line != '':
            list = eval(line)
            lists.append(list)

    sort(lists)
    ans = 1
    for i, list in enumerate(lists):
        if list == [[2]] or list == [[6]]:
            print(f'{i+1}: {list}')
            ans *= i + 1

    print(f'ans: {ans}')
        

if __name__ == "__main__":
    main()